### PR TITLE
Added feature to check for issue template mismatch

### LINKED ISSUE #76: Check if issues match provided template
## Description
As a developer,
I need to add a check on new issues,
so that the bot will label template mismatched issues.

## Mocks
-N.A.-

### Update [Required]
- [ ] The bot should label issues that don't follow the templates with 'Template Mismatch' label.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by sammy1997**
# Description
Added a check on newly opened issues so that issues not following provided templates get labelled 'Template Mismatch'.

Fixes #76 

# Type of Change:
- Code
- This change requires a documentation update
- New feature (a non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
When issue is according to template:
template_match

When issue template isn't followed:
issue_template_mismatch_1
template_mismatch_2

# Checklist:
- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: github_functions.py [hunk 1]
```diff
@@ -189,3 +187,36 @@ def close_pr(repo_owner, repo_name, pr_number):
...
+def are_issue_essential_components_present(body):
+    tokens = body.split('\r\n')
+    #Remove blank strings
+    tokens = [s for s in tokens if s != '']
+    #Necessary components in the template
+    necessary_elements_set = {'## Description', '## Acceptance Criteria', '### Update [Required]',
+                              '## Definition of Done', '## Estimation'}
+    if set(tokens).intersection(necessary_elements_set) == necessary_elements_set:
+        #Check if the template format has been followed and contents under any header isn't empty
+        if tokens[tokens.index('## Description') + 1] != '## Mocks' and tokens[tokens.index('## Description') + 1] != \
```
Comment (sara-02) [inline github_functions.py]: I am a little confused with some many conditions. You are checking that after Description either Mocks or Acceptance criteria should be there. Why not use a `or` condition here.
```
condition1 =  tokens[tokens.index('## Description') + 1] == '## Mocks' or tokens[tokens.index('## Description') + 1] == '## Acceptance Criteria'
condition2 = tokens[tokens.index('## Acceptance Criteria') + 1] == '### Update [Required]
condition3 = tokens[tokens.index('### Update [Required]') + 1] == '## Definition of Done' or
tokens[tokens.index('### Update [Required]') + 1] == '### Enhancement to Update [Optional]'
condition4 = tokens[tokens.index('## Definition of Done') + 1] != '## Estimation'
if condition 1 and condition 2 and condition 3 and condition 4 and tokens[-1] == '## Estimation'
      return True
```

Comment (sara-02) [reply github_functions.py]: Also, we can avoid the last two else conditions and simply return a False. That should also work

Comment (sammy1997) [reply github_functions.py]: @sara-02 The 'and' should be there as I want the next line to be neither of the two. Or would mean if either of them isn't present then it gets classified right, but actually neither should be there.

Comment (sara-02) [reply github_functions.py]: Ah. Ok. I follow it now :P

File: main_server.py [hunk 2]
```diff
@@ -24,7 +24,14 @@ def github_hook_receiver_function():
...
                 #If it's an issue opened event and not PR opened event
-                response = label_opened_issue(data)
+                issue_number = data.get('issue', {}).get('number', '')
+                repo_name = data.get('repository', {}).get('name', '')
+                repo_owner = data.get('repository', {}).get('owner', {}).get('login', '')
+                issue_body = data.get('issue', {}).get('body', '')
+                #Label newly opened issue
+                label_opened_issue(data)
+                #Check if opened issue follows issue template
+                check_issue_template(repo_owner, repo_name, issue_number, issue_body)
```
[the remark this answers is not in the document]
Comment (sara-02) [reply main_server.py]: Oh. OK I guess we are good then :)

**Pull request merged**